README.md
# MAPintosh.de

This is the source of https://mapintosh.de/

# License

## creative commons

# CC0 1.0 Universal

CREATIVE COMMONS CORPORATION IS NOT A LAW FIRM AND DOES NOT PROVIDE LEGAL SERVICES. DISTRIBUTION OF THIS DOCUMENT DOES NOT CREATE AN ATTORNEY-CLIENT RELATIONSHIP. CREATIVE COMMONS PROVIDES THIS INFORMATION ON AN "AS-IS" BASIS. CREATIVE COMMONS MAKES NO WARRANTIES REGARDING THE USE OF THIS DOCUMENT OR THE INFORMATION OR WORKS PROVIDED HEREUNDER, AND DISCLAIMS LIABILITY FOR DAMAGES RESULTING FROM THE USE OF THIS DOCUMENT OR THE INFORMATION OR WORKS PROVIDED HEREUNDER.

### Statement of Purpose

The laws of most jurisdictions throughout the world automatically confer exclusive Copyright and Related Rights (defined below) upon the creator and subsequent owner(s) (each and all, an "owner") of an original work of authorship and/or a database (each, a "Work").

Certain owners wish to permanently relinquish those rights to a Work for the purpose of contributing to a commons of creative, cultural and scientific works ("Commons") that the public can reliably and without fear of later claims of infringement build upon, modify, incorporate in other works, reuse and redistribute as freely as possible in any form whatsoever and for any purposes, including without limitation commercial purposes. These owners may contribute to the Commons to promote the ideal of a free culture and the further production of creative, cultural and scientific works, or to gain reputation or greater distribution for their Work in part through the use and efforts of others.

For these and/or other purposes and motivations, and without any expectation of additional consideration or compensation, the person associating CC0 with a Work (the "Affirmer"), to the extent that he or she is an owner of Copyright and Related Rights in the Work, voluntarily elects to apply CC0 to the Work and publicly distribute the Work under its terms, with knowledge of his or her Copyright and Related Rights in the Work and the meaning and intended legal effect of CC0 on those rights.

1. __Copyright and Related Rights.__ A Work made available under CC0 may be protected by copyright and related or neighboring rights ("Copyright and Related Rights"). Copyright and Related Rights include, but are not limited to, the following:

    i. the right to reproduce, adapt, distribute, perform, display, communicate, and translate a Work;

    ii. moral rights retained by the original author(s) and/or performer(s);

    iii. publicity and privacy rights pertaining to a person's image or likeness depicted in a Work;

    iv. rights protecting against unfair competition in regards to a Work, subject to the limitations in paragraph 4(a), below;

    v. rights protecting the extraction, dissemination, use and reuse of data in a Work;

    vi. database rights (such as those arising under Directive 96/9/EC of the European Parliament and of the Council of 11 March 1996 on the legal protection of databases, and under any national implementation thereof, including any amended or successor version of such directive); and

    vii. other similar, equivalent or corresponding rights throughout the world based on applicable law or treaty, and any national implementations thereof.

2. __Waiver.__ To the greatest extent permitted by, but not in contravention of, applicable law, Affirmer hereby overtly, fully, permanently, irrevocably and unconditionally waives, abandons, and surrenders all of Affirmer's Copyright and Related Rights and associated claims and causes of action, whether now known or unknown (including existing as well as future claims and causes of action), in the Work (i) in all territories worldwide, (ii) for the maximum duration provided by applicable law or treaty (including future time extensions), (iii) in any current or future medium and for any number of copies, and (iv) for any purpose whatsoever, including without limitation commercial, advertising or promotional purposes (the "Waiver"). Affirmer makes the Waiver for the benefit of each member of the public at large and to the detriment of Affirmer's heirs and successors, fully intending that such Waiver shall not be subject to revocation, rescission, cancellation, termination, or any other legal or equitable action to disrupt the quiet enjoyment of the Work by the public as contemplated by Affirmer's express Statement of Purpose.

3. __Public License Fallback.__ Should any part of the Waiver for any reason be judged legally invalid or ineffective under applicable law, then the Waiver shall be preserved to the maximum extent permitted taking into account Affirmer's express Statement of Purpose. In addition, to the extent the Waiver is so judged Affirmer hereby grants to each affected person a royalty-free, non transferable, non sublicensable, non exclusive, irrevocable and unconditional license to exercise Affirmer's Copyright and Related Rights in the Work (i) in all territories worldwide, (ii) for the maximum duration provided by applicable law or treaty (including future time extensions), (iii) in any current or future medium and for any number of copies, and (iv) for any purpose whatsoever, including without limitation commercial, advertising or promotional purposes (the "License"). The License shall be deemed effective as of the date CC0 was applied by Affirmer to the Work. Should any part of the License for any reason be judged legally invalid or ineffective under applicable law, such partial invalidity or ineffectiveness shall not invalidate the remainder of the License, and in such case Affirmer hereby affirms that he or she will not (i) exercise any of his or her remaining Copyright and Related Rights in the Work or (ii) assert any associated claims and causes of action with respect to the Work, in either case contrary to Affirmer's express Statement of Purpose.

4. __Limitations and Disclaimers.__

    a. No trademark or patent rights held by Affirmer are waived, abandoned, surrendered, licensed or otherwise affected by this document.

    b. Affirmer offers the Work as-is and makes no representations or warranties of any kind concerning the Work, express, implied, statutory or otherwise, including without limitation warranties of title, merchantability, fitness for a particular purpose, non infringement, or the absence of latent or other defects, accuracy, or the present or absence of errors, whether or not discoverable, all to the greatest extent permissible under applicable law.

    c. Affirmer disclaims responsibility for clearing rights of other persons that may apply to the Work or any use thereof, including without limitation any person's Copyright and Related Rights in the Work. Further, Affirmer disclaims responsibility for obtaining any necessary consents, permissions or other rights required for any use of the Work.

    d. Affirmer understands and acknowledges that Creative Commons is not a party to this document and has no duty or obligation with respect to this CC0 or use of the Work.


changelog.txt
2021-05-09 added screenshots
2021-05-08 index page now uses twig
2020-06-10 added 'Stolpersteine in Bonn'
2020-06-02 added 'Recycling in Bonn'
2020-05-31 added 'Trees in Bonn'
2020-05-30 selected districts of Bonn are shown in color, everything else is dimmed out
2020-05-29 use minimized files
2020-05-27 Local, pre-rendered tiles
2020-05-26 Added grayscale rendering to maps
2020-05-25 Maps now have maximize buttons
           'OSM Map of Porto' now uses layer selector
2020-05-24 First version of the site


pageSetup.php
<?php
$pageSetup = [
    'screenshotPath' => 'images/screenshots/',
    'thumbnailPath' => 'images/thumbnails/',
    'content' => [
        [
            'title' => 'Online Maps',
            'description' => [
                '<p>',
                'Some online maps I created',
                '</p>'
            ],
            'tableId' => 'webmaps',
            'tableContent' => [
                [
                    'map' => [
                        'path' => 'BN-districts',
                        'title' => 'Districts of Bonn'
                    ],
                    'modified' => '2020-05-30',
                    'created' => '2020-05-29',
                    'description' => [
                        'city districts limits are taken from <a href="https=>//opendata.bonn.de">opendata.bonn.de</a> and then processed by a ',
                        '<a href="https=>//github.com/penpendede/mapintosh.de/blob/master/BN-districts/getDistricts.sh">tiny shell script</a> you may find interesting.'
                    ],
                    'screenshot'=> 'BN-districts.jpg',
                    'thumbnail'=> 'BN-districts.jpg',
                    'changes' => [
                        'selected districts are shown in color, everything else is dimmed out'
                    ]
                ],
                [
                    'map' => [
                        'path' => 'airportWeather/website',
                        'title' => 'METAR (weather) data for airports'
                    ],
                    'modified' => '2020-09-12',
                    'created' => '2020-09-13',
                    'description' => [
                        'provides access to wether information for  airports worldwide. The source of this project is available at ',
                        '<a href="https=>//github.com/penpendede/airportWeather" target="_blank">github.com/penpendede/airportWeather</a>'
                    ],
                    'screenshot'=> 'airportWeather.jpg',
                    'thumbnail'=> 'airportWeather.jpg',
                    'changes' => [
                        'now using a git submodule for airportWeather'
                    ]
                ],
                [
                    'map' => [
                        'path' => 'myOsm',
                        'title' => 'OSM Map of Bonn'
                    ],
                    'modified' => '2020-05-29',
                    'created' => '2020-05-24',
                    'description' => [
                        'accompanies ',
                        '<a href="http=>//penpende.de/doku.php?id=installation_guides=>tileserver_on_raspberry_pi_4" target="_blank">Installing a tile server on a Raspberry Pi 4</a>'
                    ],
                    'screenshot'=> 'myOsm.jpg',
                    'thumbnail'=> 'myOsm.jpg',
                    'changes' => [
                        'use minimized files',
                        'local, pre-rendered tiles',
                        'added grayscale rendering',
                        'maximize button'
                    ]
                ],
                [
                    'map' => [
                        'path' => 'Porto',
                        'title' => 'Porto'
                    ],
                    'modified' => '2020-05-29',
                    'created' => '2020-05-24',
                    'description' => [
                        'accompanies <a href="http=>//penpende.de/doku.php?id=gis=>making_use_of_openstreetmap" target="_blank">Making use of OpenStreetMap</a>'
                    ],
                    'screenshot'=> 'Porto.jpg',
                    'thumbnail'=> 'Porto.jpg',
                    'changes' => [
                        '<ul>',
                        '<li>use minimized files</li>',
                        '<li>added grayscale rendering</li>',
                        '<li>maximize button</li>',
                        '<li>instead of showing all kinds of POI you can now select it using the layer selector in the upper right corner.</li>',
                        '</ul>'
                    ]
                ],
                [
                    'map' => [
                        'path' => 'BN-recycling',
                        'title' => 'Recycling in Bonn'
                    ],
                    'modified' => '2020-06-02',
                    'created' => '2020-06-02',
                    'description' => [
                        'Part of getData.js deals with fixing issues with the original data. Once they are fixed the original script will remain available as it ',
                        'illustrates how command-line tools can be used to fix certain issues with GeoJSON files.'
                    ],
                    'screenshot'=> 'BN-recycling.jpg',
                    'thumbnail'=> 'BN-recycling.jpg',
                    'changes' => [
                        '-'
                    ]
                ],
                [
                    'map' => [
                        'path' => 'stolpersteine',
                        'title' => 'all stolpersteins from the OSM database'
                    ],
                    'modified' => '2020-11-15',
                    'created' => '2020-11-10',
                    'description' => [
                        'an overview of the thousands of stolperteins that have already been added to the OSM database'
                    ],
                    'screenshot'=> 'stolpersteine.jpg',
                    'thumbnail'=> 'stolpersteine.jpg',
                    'changes' => [
                        'Using tab separated values instead of geoJSON, considerably reducing the amount of data that needs to be transferred.'
                    ]
                ],
                [
                    'map' => [
                        'path' => 'BN-tree',
                        'title' => 'Trees in Bonn'
                    ],
                    'modified' => '2020-05-31',
                    'created' => '2020-05-31',
                    'description' => [
                        'Visualizes trees in public space (streets, parks) within the city limits of Bonn and provides the species as well as the age (open data, ',
                        'over 60k entries). It also shows how you can visualize such a huge dataset and still have Leaflet render it very rapidly. It uses open data ',
                        'from  <a href="https=>//opendata.bonn.de/dataset/baumstandorte" target="_blank">https=>//opendata.bonn.de/dataset/baumstandorte</a> released ',
                        'under CC0'
                    ],
                    'screenshot'=> 'BN-tree.jpg',
                    'thumbnail'=> 'BN-tree.jpg',
                    'changes' => [
                        '-'
                    ]
                ]
            ]
        ],
        [
            'title' => 'Maps with QGIS',
            'description' => [
                '<p>',
                '<a href="https=>//www.qgis.org/" target="_blank">QGIS</a> is a free and open source geographic information system for the desktop that is available for ',
                'many platforms.',
                '</p>'
            ],
            'tableId' => 'qgismaps',
            'tableContent' => [
                [
                    'map' => [
                        'path' => null,
                        'title' => 'Stolpersteine in Bonn'
                    ],
                    'modified' => '2020-06-10',
                    'created' => '2020-06-10',
                    'description' => [
                        '<p>',
                        'A QGIS project that has all stolpersteine in Bonn. The GeoPackage file was generated using QGIS and a cute plugin that queries the OSM database. ',
                        'Description how to generated it will follow. For now you can',
                        '</p>',
                        '<ul>',
                        '<li>',
                        '<a href="QGIS/Stolpersteine_in_Bonn/Stolpersteine_in_Bonn.qgs" target="_blank">Get the qgis project</a> (data is loaded from this server)',
                        '</li>',
                        '<li>',
                        '<a href="QGIS/Stolpersteine_in_Bonn/Stolpersteine_in_Bonn_Data.zip" target="_blank">Get a ZIP</a> that contains all files used (including the',
                        'QGIS project)</li>',
                        '</ul>',
                        '<p>',
                        'You don\'t already have QGIS? No problem, it is available for most platforms at <a href="https=>//www.qgis.org/" target="_blank">qgis.org</a>.',
                        '</p>'
                    ],
                    'screenshot'=> '',
                    'thumbnail' => '',
                    'changes' => [
                        '-'
                    ]
                ]
            ]
        ]
    ]
];
